receiver/purefareceiver/internal/scraper.go

```diff
@@ -71,17 +77,20 @@ func (h *scraper) ToPrometheusReceiverConfig(host component.Host, _ receiver.Fac
 
 		httpConfig := configutil.HTTPClientConfig{}
 		httpConfig.BearerToken = configutil.Secret(bearerToken)
+		httpConfig.TLSConfig.InsecureSkipVerify = h.insecureskipverify
```

Comment:
Interesting, what if people set the other options from tls config, are they going to be ignored? You could probably pass the entire TLSConfig from the config to the scraper, no? In fact, we could probably just reuse the entire client config here...

Reply:
This may be out of my skill range to do that. I tried passing the reciever tlsconfig directly to the Prometheus TLSConfig, and they were incompatible.  I'll try and poke around some more for a cleaner way to do this. Please let me know if there's any hints you can give me. thank you!

Reply:
You are right -- they are similar, but of the same type. Perhaps something like:

```go
type scraper struct {
	scraperType    ScraperType
	endpoint       string
	namespace      string
	tlsSettings    configtls.ClientConfig
	configs        []ScraperConfig
	scrapeInterval time.Duration
	labels         model.LabelSet
}
...
func (h *scraper) ToPrometheusReceiverConfig(host component.Host, _ receiver.Factory) ([]*config.ScrapeConfig, error) {
...
		httpConfig.TLSConfig = configutil.TLSConfig{
			InsecureSkipVerify: h.tlsSettings.InsecureSkipVerify,
			ServerName:         h.tlsSettings.ServerName,
			CAFile:             h.tlsSettings.CAFile,
			CertFile:           h.tlsSettings.CertFile,
			KeyFile:            h.tlsSettings.KeyFile,
			...
		}
...
}
```

The goal is to provide a way that is consistent across components, while setting the right expectations to users. In your case, the UX is consistent, as Collector users are exposed to the same config used in other components, but if they set `MinVersion`, this isn't going to be respected.

I'd recommend to either log a warning if anything else other than the `InsecureSkipVerify` flag is set, or to support the most options possible (which include more than what I added there explicitly).

Reply:
Some of these will be harder to implement than others - eg. MinVersion for the prom http config is a uint16, I've otherwise mapped the rest of the struts. 

Let me know what you think, and i'll also test with an array today.